src/js/components/validateData/ValidateDataFileComponent.jsx

```diff
@@ -222,6 +225,72 @@ export default class ValidateDataFileComponent extends React.Component {
         return icon;
     }
 
+    signReport(item){
+        let type = null;
+        switch(item.file_type){
+            case 'appropriations':
+                type='A';
+                break;
+            case 'program_activity':
+                type='B';
+                break;
+            case 'award_financial':
+                type='C';
+                break;
+            case 'detached_award':
+                type='D2_detached';
+                break;
+            default:
+                break;
+        }
+        if(type){
+            GenerateFilesHelper.fetchFile(type, this.props.submission.id)
+            .then((result)=>{
+                this.setState({
+                    signInProgress: false,
+                    signedUrl: result.url
+                }, () => {
+                    this.openReport();
+                });
+            })
+            .catch((err) => {
+                this.setState({
+                    signInProgress: false
+                });
+                console.log(err);
+            });    
+        }else{
```

Comment:
Be consistent with your spacing. We usually put a space after the `else` so it looks like `else {` and also after if statements `if(truth check) {`. Also, stay consistent with where the else goes. MOST of the time, we do:
```
}
else {
```